gbpservice/nfp/configurator/lib/schema_validator.py

```diff
@@ -0,0 +1,77 @@
+from oslo_log import log
+import gbpservice.nfp.configurator.lib.schema as schema
+
+LOG = log.getLogger(__name__)
+
+
+class SchemaValidator():
+
+    def decode(self, request_data):
+        try:
+            if not self.validate_schema(request_data, schema.request_data):
+                return False
+
+            if ('service_type' in request_data['info'] and
+                    'version' in request_data['info']):
+                service_type = request_data['info']['service_type']
+            elif ('service_type' not in request_data['info'] and
+                    'version' in request_data['info']):
+                service_type = 'generic'
```

Comment:
Since the schema validation is dependent on the service_type it is better to validate the service type. We can get invalid service type. Earlier, this was validated in configurator but that block got removed now.